generator/views/ping_view.py

```diff
@@ -43,8 +43,8 @@ def from_db_views(
                 if channel.get("channel") is not None:
                     table["channel"] = channel["channel"]
                 if (
-                    len(references) != 1
-                    or references[0][-2] != channel["source_dataset"]
+                    len(set(r[-1] for r in references)) != 1
+                    or channel["source_dataset"] not in [r[-2] for r in references]
                 ):
```

Comment:
The Black format CI check is failing because it wants to format this like
```python
                if len(set(r[-1] for r in references)) != 1 or channel[
                    "source_dataset"
                ] not in [r[-2] for r in references]:
```
which I think is objectively much worse for code readability.

I'll see if I can find some way to satisfy the Black formatter while maintaining readability.